### ISSUE TITLE
Combine spans from Gradle into a parent Jenkins trace

### ISSUE THREAD

**Issue by jacob-bresciani-everbridge** [craigatk/opentelemetry-gradle-plugin#97]
I'm looking at using this plugin but having the gradle build triggered from a Jenkins job which is using the [Jenkins opentelemetry plugin](https://plugins.jenkins.io/opentelemetry/). Unfortunately this results in a trace for the Jenkins job and a separate trace for the gradle task. Is it possible to have the gradle spans show up in Jenkins job's trace?

**Comment by craigatk** [comment]
Good question! From [this comment in the Jenkins plugin](https://plugins.jenkins.io/opentelemetry/#plugin-content-opentelemetry-maven-extension), it sounds like there is a way to have the Jenkins plugin export its OpenTelemetry info as environment variables - and then the Gradle plugin could read those.

I think I'd just need the names of the environment variables that the Jenkins plugin uses for the trace ID and span ID, and then I can have the Gradle plugin attach those as the parent - and then the Gradle spans should hopefully combine together into a singe trace with the Jenkins plugin.

**Comment by jacob-bresciani-everbridge** [comment]
it sets the following that I am aware of

SPAN_ID=f1a2153e247b0d94
TRACEPARENT=00-a263fdf001993a32980b9ec5740b7d6d-f1a2153e247b0d94-01
TRACE_ID=a263fdf001993a32980b9ec5740b7d6d

plus a few others you might be interested in (values based on plugin config)

OTEL_EXPORTER_OTLP_ENDPOINT=http://collector.foo.bar:4317/
OTEL_EXPORTER_OTLP_INSECURE=true/false (depending on how you have it configured)
OTEL_TRACES_EXPORTER=otlp

**Comment by craigatk** [comment]
Thanks for the environment variable names!

I released a new version of the plugin `1.7.0` that aims to use the `SPAN_ID` and `TRACE_ID` from remote parents like Jenkins to connect the spans from this plugin to that parent trace.

I don't have a Jenkins instance to fully test it with myself, but when you get a chance please give the new plugin version a try and let me know if it works for you :)

**Comment by jacob-bresciani-everbridge** [comment]
Thanks for the work!

Unfortunately it doesn't work as expected.

I've added some echo statements to jenkins and println to gradle and I get the following output 

```
[Pipeline] echo
---------------- Jenkins - env.TRACE_ID: bbcffc15af0e0577b8fa3ea0d699cfbe, env.SPAN_ID: 067021354dccd242 ----------------
[Pipeline] dir
Running in /tmp/src
[Pipeline] {
[Pipeline] sh (gradlew)
+ ./gradlew jibBuildTar --profile -S --no-daemon
Downloading https://services.gradle.org/distributions/gradle-8.2.1-bin.zip
............10%............20%............30%.............40%............50%............60%............70%.............80%............90%............100%

Welcome to Gradle 8.2.1!

Here are the highlights of this release:
 - Kotlin DSL: new reference documentation, assignment syntax by default
 - Kotlin DSL is now the default with Gradle init
 - Improved suggestions to resolve errors in console output
 - Reduced sync memory consumption

For more details see https://docs.gradle.org/8.2.1/release-notes.html

To honour the JVM settings for this build a single-use Daemon process will be forked. For more on this, please refer to https://docs.gradle.org/8.2.1/userguide/gradle_daemon.html#sec:disabling_the_daemon in the Gradle documentation.
Daemon will be stopped at the end of the build 

> Configure project :
---------------- gradle - System.env.TRACE_ID: bbcffc15af0e0577b8fa3ea0d699cfbe, System.env.SPAN_ID: f35ea553455fc10c ----------------

> Task :compileJava UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :jibBuildTar UP-TO-DATE

OpenTelemetry build trace ID acc439549dc8c210f8bd91992ccb8766

BUILD SUCCESSFUL in 27s
3 actionable tasks: 3 up-to-date
```

If you wish to see what I've setup (perhaps I've just made a gradle configuration mistake) and test it yourself I've sanitized my code and setup a docker-compose file that sets up:
- jenkins (including the opentelemetry lib and it's config)
- grafana
- tempo (acting as the trace collector target)
- prometheus (trace metrics collection)

and pushed it to my personal repo here: https://gitlab.com/jbresciani/opentelemetry (if this works for you, feel free to copy it for yourself)

**Comment by craigatk** [comment]
Dang, sorry the change didn't work! I'll investigate and see if I can find out what's going wrong.

And thanks so much for doing all the work to put together that example repo - that's going way above and beyond and I really appreciate it :)

**Comment by craigatk** [comment]
I think I've got a fix for this! And testing with your Jenkins example helped :)

I released new plugin version 1.7.1 with the fix, please let me know how it works for you :)
